src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -202,6 +206,15 @@ func (r *StatefulSetResource) rollingUpdate(
 		if err = r.queryRedpandaStatus(ctx, &adminURL); err != nil {
 			return fmt.Errorf("unable to query Redpanda ready status: %w", err)
 		}
+
+		adminURL.Path = "metrics"
+
+		if err = r.queryRedpandaUnderReplicatedPartition(ctx, &adminURL); err != nil {
+			return &RequeueAfterError{
+				RequeueAfter: RequeueDuration,
+				Msg:          fmt.Sprintf("under replicated partition is not ready: %v", err),
```

Comment:
Maybe a better description of the error...

Reply:
I will try with `broker reported under replicated partitions: %v`